lib/sort_by_length.rb

```diff
@@ -1,7 +1,33 @@
 # A method which will return an array of the words in the string
 #  sorted by the length of the word.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n2)
+# Space complexity: O(1)
 def sort_by_length(my_sentence)
```

Comment:
Since you are splitting the string into an array of words, this has O(n) space complexity.  Otherwise nice implementation of InsertionSort.